code/__HELPERS/game.dm

```diff
@@ -484,24 +485,55 @@
 
 	return result
 
-/proc/pollCandidatesForMob(Question, jobbanType, role_preference_key, poll_time = 300, mob/M, ignore_category = null)
-	var/list/L = pollGhostCandidates(Question, jobbanType, role_preference_key, poll_time, ignore_category)
-	if(QDELETED(M) || !M.loc)
+/**
+ * Returns a list of ghosts that are eligible to take over and wish to be considered for a mob.
+ *
+ * Arguments:
+ * * question - Question to show players as part of poll
+ * * jobban_type - Type of jobban to use to filter out potential candidates.
+
+ * * poll_time - Length of time in deciseconds that the poll input box exists before closing.
+ * * target_mob - The mob that is being polled for.
+ * * ignore_category - Unknown/needs further documentation.
+ */
+/proc/poll_candidates_for_mob(question, jobban_type, role_preference_key, poll_time = 30 SECONDS, mob/target_mob, ignore_category = null)
+	var/static/list/mob/currently_polling_mobs = list()
```

Comment:
why not just eliminate the source of the spam?

This means if multiple polls happen within the same 30sec or so, they will only ever see the first one. If something is specifically susceptible to being spammed, it should be reduced. This will just break edge cases where say a midround spawns right after someone uses a sentience potion, and now that midround does not spawn at all.

Reply:
point taken

Reply:
reverted